lib/yocaml/metadata.ml

```diff
@@ -6,10 +6,18 @@ module type INJECTABLE = sig
   val to_mustache : t -> (string * Mustache.Json.value) list
 end
 
+module type PROVIDER = sig
+  type t
+
+  val from_string : string -> t Validate.t
+
+  include Key_value.KEY_VALUE_VALIDATOR with type t := t
+end
+
 module type PARSABLE = sig
   type t
 
-  val from_string : string option -> t Validate.t
+  val from_string : (module PROVIDER) -> string option -> t Validate.t
 end
```

Comment:
I think that PROVIDER should be PARSABLE (or VALIDABLE) and PARSABLE (my bad!) should be READABLE (or something more symetric to INJECTABLE)

Reply:
Yes, I go for VALIDABLE and READABLE.

Reply:
Yes it is better!